cylc/flow/task_events_mgr.py

```diff
@@ -1300,6 +1300,15 @@ def _retry_task(self, itask, wallclock_time, submit_retry=False):
             )
             itask.state.add_xtrigger(label)
 
+        # add the retry xtrigger to the data store
+        sig = self.xtrigger_mgr.get_xtrig_ctx(itask, label).get_signature()
+        (
+            self.data_store_mgr.xtrigger_tasks
+            .setdefault(sig, set())
+            .add((itask.tokens.id, label))
+        )
+        self.data_store_mgr.delta_task_xtrigger(sig, False)
```

Comment:
@hjoliver, I suspect this code block resolves the issue of retry xtriggers not always appearing.

Reply:
This change probably closes https://github.com/cylc/cylc-flow/issues/6742, TODO, confirm this.

Reply:
Yes, this does close #6742.

Screenshot from `cylc show` on this branch showing a task which is waiting on its third retry:

![Screenshot from 2025-05-07 12-48-26](https://github.com/user-attachments/assets/8981336d-8c86-4f70-b8ba-224e059d91de)